functions/go/apply-setters/README.md

```diff
@@ -0,0 +1,94 @@
+# apply-setters
+
+### Overview
+
+<!--mdtogo:Short-->
+
+Apply setter values on resources fields. May set either the complete or partial field value.
+
+<!--mdtogo-->
+
+Setters provide a solution for template-free setting of field values. They are a
+safer alternative to other substitution techniques which do not have the context
+of the structured data. Setters may be invoked to modify the configuration
+using `apply-setters` function to set values.
+
+### Synopsis
+
+<!--mdtogo:Long-->
+
+```
+kpt fn eval apply-setters:VERSION [DIR] -- [setter_name=setter_value]
```

Comment:
Related to standardizing: This document doesn't actually describe the function ConfigMap directly.

This also ties functions to kpt CLI (Other funcition usage docs don't do this) whereas a function can technically be invoked by any orchestrator, not just kpt. We should discuss pros/cons of this.

Reply:
This also just shows the imperative way of running it.

I think usage docs should just document the function config and behavior. How to run functions using kpt CLI are out of scope and will be covered in detail in kpt docs.

Reply:
Agree. Updated the doc.

Reply:
These docs still show kpt CLI and talk about imperative vs declarative usage, etc. Did we settle on what's in scope for usage docs?

Reply:
I think usage docs should highlight the recommended workflow(declarative or imperative) the function should be used along with pointing to the alternative approach. In search and replace docs, I just talked about imperative flow as it is highly unlikely it will be used in declarative workflow. Let me know your thoughts.